### PR TITLE
`SDK/KeyVault/v7.3` -  Self generated data plane SDK

### PR DESCRIPTION
**Pull request by aristosvo**
First try with incorporating KV Rotation Policies based on self generated Track 1 SDK

- [x] Fixes dependency problem with #14471 (Acctests runned fine)

### CODE REVIEW

**Comment by aristosvo** [comment]
Splitted off #18603 to a separate PR

File: internal/services/keyvault/sdk/v7.3/generate.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+AZURERM_LOCATION="${GIT_REPO_LOCATION}/terraform-provider-azurerm"
+SDK_LOCATION="${AZURERM_LOCATION}/internal/services/keyvault/sdk/v7.3/keyvault"
+
+mkdir -p "${SDK_LOCATION}"
+cp "${SDK_LOCATION}/../autorest.md" "${DATAPLANE_LOCATION}"/readme.go.md
+cd "${DATAPLANE_LOCATION}/" || exit 1
+
+# Conflict between ActionType enum and ActionType type..
+# Mac specific sed? :(
+sed -i '' 's/"ActionType"/"ActionsType"/g' Microsoft.KeyVault/stable/7.3/keys.json
```
[the remark this answers is not in the document]
Comment (aristosvo) [reply internal/services/keyvault/sdk/v7.3/generate.sh]: Implemented the directive on this PR for now, what to do with Kermit is up to you :)

**Comment by wuxu92** [comment]
@tombuildsstuff it seems that Azure/azure-rest-api-specs will not fix this duplicate const type definition. would this PR be merged recently? or could we use the v7.2 version of it? @aristosvo

**Comment by tombuildsstuff** [comment]
@wuxu92 

> @tombuildsstuff it seems that Azure/azure-rest-api-specs will not fix this duplicate const type definition. would this PR be merged recently? or could we use the v7.2 version of it? @aristosvo

This is currently waiting on the PR on the Swagger repo which requires our attention, I need to dig into it since I believe the proposed changes will make things considerably more complicated for very little benefit (by requiring the use/update of multiple different SDKs, vs just providing a single SDK as is today in Track1) - I'll take a look as soon as I get a chance, but in the interim merging the Swagger PR will unblock this.

**Comment by wuxu92** [comment]
Since the swagger repo has closed the PR, will this PR be merged with that workaround?

**Comment by tombuildsstuff** [comment]
@wuxu92 we're still waiting on a reply in the Swagger repository

**Comment by wuxu92** [comment]
are we still waiting for swagger update?

**Comment by tombuildsstuff** [comment]
@wuxu92 yup, the Azure SDK team closed it without addressing the issue, I commented here earlier: https://github.com/Azure/azure-rest-api-specs/pull/21137#discussion_r1021084427

**Comment by tombuildsstuff** [comment]
@aristosvo apologies for the delay getting back to you on this one - the Swagger changes have been merged into the release branch for v7.4 of KeyVault Data Plane API's - I've [updated this PR on `tombuildsstuff/kermit`](https://github.com/tombuildsstuff/kermit/pull/1) in preparation for the Key Vault 7.4 PR being merged into `main`, once that's done we can merge that PR and vendor it (as we are for the other Track1 SDK's) rather than inlining this SDK.

As such I hope you don't mind but I'm going to close this PR in favour of https://github.com/tombuildsstuff/kermit/pull/1 - but once the Key Vault Data Plane v7.4 changes have landed in `main` in `Azure/azure-rest-api-specs` (which we're keeping an eye out for), we can look to proceed with #18603.

Thanks!

**Comment by aristosvo** [comment]
No problem, thanks for keeping it on the radar anyway! Looking forward to land the features 🍻

**Pull request closed without merging**